### PR TITLE
vendor/rename protobuf package

### PR DESCRIPTION
**Pull request by jrobsonchase**
This is a copy of what we use internally, but with our extensions (that just rename fields) stripped out and with a new package name to avoid protoreflect name collisions.

### CODE REVIEW

**Comment by jrobsonchase** [comment]
Also ran the online/authenticated/paid tests with this change, and everything still passes :tada:

File: scripts/Makefile [hunk 1]
```diff
@@ -15,3 +15,6 @@ docs:
 	pkill pkgsite || true
 	pkgsite >/dev/null 2>&1 & \
 		xdg-open http://127.0.0.1:8080/golang.ngrok.com/ngrok
+
+gen-proto:
+	protoc --go_out=. --go_opt=paths=source_relative internal/pb_agent/middleware.proto
```
[the remark this answers is not in the document]
Comment (jrobsonchase) [reply scripts/Makefile]: Cool, switched to that :+1:

**Review by euank — APPROVED** [review]
lgtm, ty!

**Comment by jrobsonchase** [comment]
So I would merge, but...

The fmt/gen jobs don't actually test anything :confused: Either no files are being updated or the `git diff --exit-code` isn't seeing them for some reason.

**Comment by jrobsonchase** [comment]
I'm gonna go ahead and merge this and file an issue for the bad diff checks, since the problem predates this change. We can fix both at the same time in a follow-up.

**Pull request merged**